src/sqlfluff/rules/L036.py

```diff
@@ -333,5 +369,33 @@ def _eval_single_select_target_element(
                 anchor=select_clause,
                 fixes=fixes,
             )
-        else:
-            return None
+
+        # If we have a wildcard on the same line as the FROM keyword, but not the same line
+        # as the SELECT keyword, we need to move the FROM keyword to its own line.
+        # i.e.
+        # SELECT
+        #   * FROM foo
+        if select_targets_info.from_segment:
+            if (
+                is_wildcard
+                and (
+                    select_clause.pos_marker.working_line_no
+                    != select_targets_info.from_segment.pos_marker.working_line_no
+                )
+                and (
+                    wildcard_select_clause_element.pos_marker.working_line_no
+                    == select_targets_info.from_segment.pos_marker.working_line_no
+                )
+            ):
+                fixes = [
```

Comment:
Is this the same code as line 147? I wonder if we ought to create a function with this code and call it from both these places.

Reply:
So my thinking on this was:
On lines 142-157, we are appending these lint fixes to an already existing lint fixes, whereas on lines 378-399 this condition only a wildcard and creates a separate list of fixes to the rest of the code. Given that the conditions in the if clause are also different, the only thing we could add to a function would be the
```python
 fixes.extend(
                        [
                            LintFix("delete", ws)
                            for ws in select_targets_info.pre_from_whitespace
                        ]
                    )
                    fixes.append(
                        LintFix(
                            "create", select_targets_info.from_segment, NewlineSegment()
                        )
                    )
```
lines, which IMO are already pretty self explanatory so maybe not much is gained in terms of moving them to a separate method?

Happy to implement the change if you think it's worth it though 😄